.github/workflows/backport_branches.yml

```diff
@@ -349,6 +349,13 @@ jobs:
         with:
           clear-repository: true
           submodules: true
+      - name: Apply sparse checkout for contrib # in order to check that it doesn't break build
```

Comment:
I suggest reverting this part in the CI. 

Besides the fact, that it only does the additional work after the modules are already checked out, this part has broken the debug build https://github.com/ClickHouse/ClickHouse/actions/runs/5033995222/jobs/9028616459#step:7:3663

cc @nickitat

Reply:
@Felixoid, that's exactly why I added this to the CI. Changes in https://github.com/ClickHouse/ClickHouse/pull/50037 break sparse checkout, so it must be fixed in https://github.com/ClickHouse/ClickHouse/pull/50037 before merging.

Reply:
it makes sense to check this script in CI, but maybe as a separate step with self-explanatory name. otherwise people will just waste time over and over again because obviously most of them have no clue about this logic. and they shouldn't.

Reply:
Agree with Nikina. It's unclear now what is broken there, and why it differs from other steps

Reply:
Yes, it was unobvious that we use sparse checkout for debug build only, sorry for that. But I don't like introducing another build just to check that sparse checkout works, because it will actually duplicate an ordinary build for no good reason. I would prefer to add some note about this to the report. 

> we should use one of the fastest builds instead of a long one

It does not matter, sparse checkout does not make a build slower, it just takes a few seconds to re-checkout submodules, and after that build continues as usual.

> What do you think of BuilderBinRelease as the job to test the sparse checkout?

Release build is slower than debug build.

Reply:
Release deb yes, but I am talking regarding the release binary build. See the [report](https://s3.amazonaws.com/clickhouse-test-reports/0/cc3897a84aa607d172f593539c22153f200584e1/clickhouse_build_check/report.html), 0:16:20 for the BuilderBinRelease vs 0:20:05 for BuilderDebDebug

I am speaking not about adding a new check, but renaming the existing one to reflect what we actually checking there.